### PR TITLE
Replace deprecated numpy.alltrue method

### PR DESCRIPTION
**Pull request by eendebakpt**
The method `numpy.alltrue` has been deprecated in 1.25 and will be removed in numpy 2.0. See https://github.com/numpy/numpy/pull/23314

### CODE REVIEW

**Review by dschult — APPROVED** [review]
Thank you!  This looks good and is one of those hard to find changes. 

I searched for this function elsewhere in networkx and this seems to be the only location.
LGTM

**Review by rossbar — APPROVED** [review]
LGTM as well - tests pass locally (with scipy 1.10) so the failures are unrelated, but I'll hold off on the merge button in the name of prudence! Thanks @eendebakpt

**Pull request merged**
